client/blindbox/requests/secure_session.py

```diff
@@ -126,6 +135,78 @@ def request(self, method, endpoint, *args, **kwargs):
+        attest_url = ":".join(self.base_url.split(":")[:-1])
+        print(attest_url)
+        res = super().request("POST",f"{attest_url}:8080/attest/maa", json={"maa_endpoint":"blindboxattester.eus.attest.azure.net","runtime_data":b64nonce})
+        maa_token = json.loads(res.content)["token"]
+        header = maa_token.split(".")[0]
+
+        # Ensure header is appropriate length to be base64 decoded
+        if len(header)%4 != 0:
+            header += "="*(len(header)%4)
+
+        header = json.loads(base64.b64decode(header))
+        kid = header["kid"]
+        if header["jku"] != "https://blindboxattester.eus.attest.azure.net/certs":
+            print("Token not generated by blindbox attester")
+            exit()
+
+        jwt_certs = self.request("GET",header["jku"]).json()
+
+        for iter in jwt_certs["keys"]:
+            if iter["kid"] == kid:
+                x5c = iter["x5c"][0] #Wrap cert as pem
+                x5c = "-----BEGIN CERTIFICATE-----" + x5c + "-----END CERTIFICATE-----"
+
+        from cryptography.x509 import load_pem_x509_certificate
+
+        cert_obj = load_pem_x509_certificate(x5c.encode())
+        public_key = cert_obj.public_key()
+        
+        # Decodes jwt and validates signature and expiry
+        payload = jwt.decode(maa_token,public_key,algorithms=["RS256"],)
+
+        # Add further checks for issued at time (iat) and issuer (iss) 
+        if payload["x-ms-attestation-type"] != "sevsnpvm":
+            raise AttestationException("IncorrectAttestation","Attestation validation failed (not sev-snp report). Exiting.")
+            #print("Attestation validation failed (not sev-snp report). Exiting.")
+            exit()
+        if payload["x-ms-compliance-status"] != "azure-compliant-uvm":
+            raise AttestationException("NonCompliantUvm","Attestation validation failed (non-compliant uvm). Exiting.")
+            #print("Attestation validation failed (non-compliant uvm). Exiting.")
+            exit()
+        if payload["x-ms-sevsnpvm-hostdata"] != cce_policy:
+            raise AttestationException("IncorrectPolicy","Attestation validation failed (cce policy mismatch). Exiting.")
+            #print("Attestation validation failed (cce policy mismatch). Exiting.")
+            exit()
+        if payload["x-ms-runtime"] != nonce:
+            raise AttestationException("IncorrectNonce","Attestation validation failed (nonce mismatch). Exiting.")
+            #print("Attestation validation failed (nonce mismatch). Exiting.")
+            exit()
+        if payload["x-ms-sevsnpvm-is-debuggable"] == "false":
+            raise AttestationException("DebugMode","Attestation validation failed (VM is in debug mode). Exiting.")
+            #print("Attestation validation failed (VM is in debug mode). Exiting.")
+            exit()
+        
+        print("Attestation validated")
+
 
 def connect(addr: str, debug_mode: bool = False) -> SecureSession:
```

Comment:
Please add the argument to specify the cci_policy here and please document it.

Reply:
Added the cce_policy field and a new argument that accepts the MAA endpoint uri